Scheduled tests on main are failing

### Checklist

- [x] I added a descriptive title
- [x] I searched open reports and couldn't find a duplicate

### What happened?

Scheduled tests on main are failing: https://github.com/conda/conda-build/actions/runs/27508891286/job/81305281635

> [!NOTE]
> This failure was not auto reported since we do not currently have auto reporting enabled, see #6011.

Jobs:
- [windows (3.10, release, serial)](https://github.com/conda/conda-build/actions/runs/27508891286/job/81305281635#logs)
- [windows (3.14, canary, serial)](https://github.com/conda/conda-build/actions/runs/27508891286/job/81305281579#logs)

Pytest:

```
FAILED tests/test_api_build.py::test_cmake_generator[3.10-64] - conda_build.exceptions.BuildScriptException: Command '['cmd.exe', '/d', '/c', 'conda_build.bat']' returned non-zero exit status 1.
FAILED tests/test_api_build.py::test_cmake_generator[3.10-32] - conda_build.exceptions.BuildScriptException: Command '['cmd.exe', '/d', '/c', 'conda_build.bat']' returned non-zero exit status 1.
FAILED tests/test_api_build.py::test_cmake_generator[3.11-32] - conda_build.exceptions.BuildScriptException: Command '['cmd.exe', '/d', '/c', 'conda_build.bat']' returned non-zero exit status 1.
FAILED tests/test_api_build.py::test_cmake_generator[3.11-64] - conda_build.exceptions.BuildScriptException: Command '['cmd.exe', '/d', '/c', 'conda_build.bat']' returned non-zero exit status 1.
FAILED tests/test_api_build.py::test_cmake_generator[3.12-32] - conda_build.exceptions.BuildScriptException: Command '['cmd.exe', '/d', '/c', 'conda_build.bat']' returned non-zero exit status 1.
FAILED tests/test_api_build.py::test_cmake_generator[3.12-64] - conda_build.exceptions.BuildScriptException: Command '['cmd.exe', '/d', '/c', 'conda_build.bat']' returned non-zero exit status 1.
FAILED tests/test_api_build.py::test_cmake_generator[3.13-32] - conda_build.exceptions.BuildScriptException: Command '['cmd.exe', '/d', '/c', 'conda_build.bat']' returned non-zero exit status 1.
FAILED tests/test_api_build.py::test_cmake_generator[3.13-64] - conda_build.exceptions.BuildScriptException: Command '['cmd.exe', '/d', '/c', 'conda_build.bat']' returned non-zero exit status 1.
```

### Traceback

``````
________________________ test_cmake_generator[3.10-64] ________________________
[gw1] win32 -- Python 3.10.20 C:\Miniconda\envs\test\python.exe
Traceback (most recent call last):
  File "D:\a\conda-build\conda-build\conda_build\build.py", line 2541, in build
    windows.build(
  File "D:\a\conda-build\conda-build\conda_build\windows.py", line 402, in build
    check_call_env(
  File "D:\a\conda-build\conda-build\conda_build\utils.py", line 413, in check_call_env
    return _func_defaulting_env_to_os_environ("call", *popenargs, **kwargs)
  File "D:\a\conda-build\conda-build\conda_build\utils.py", line 389, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
subprocess.CalledProcessError: Command '['cmd.exe', '/d', '/c', 'conda_build.bat']' returned non-zero exit status 1.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Miniconda\envs\test\lib\site-packages\_pytest\runner.py", line 353, in from_call
    result: TResult | None = func()
  File "C:\Miniconda\envs\test\lib\site-packages\_pytest\runner.py", line 245, in <lambda>
    lambda: runtest_hook(item=item, **kwds),
  File "C:\Miniconda\envs\test\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Miniconda\envs\test\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Miniconda\envs\test\lib\site-packages\pluggy\_callers.py", line 167, in _multicall
    raise exception
  File "C:\Miniconda\envs\test\lib\site-packages\pluggy\_callers.py", line 139, in _multicall
    teardown.throw(exception)
  File "C:\Miniconda\envs\test\lib\site-packages\_pytest\logging.py", line 850, in pytest_runtest_call
    yield
  File "C:\Miniconda\envs\test\lib\site-packages\pluggy\_callers.py", line 139, in _multicall
    teardown.throw(exception)
  File "C:\Miniconda\envs\test\lib\site-packages\pluggy\_callers.py", line 53, in run_old_style_hookwrapper
    return result.get_result()
  File "C:\Miniconda\envs\test\lib\site-packages\pluggy\_result.py", line 103, in get_result
    raise exc.with_traceback(tb)
  File "C:\Miniconda\envs\test\lib\site-packages\pluggy\_callers.py", line 38, in run_old_style_hookwrapper
    res = yield
  File "C:\Miniconda\envs\test\lib\site-packages\pluggy\_callers.py", line 139, in _multicall
    teardown.throw(exception)
  File "C:\Miniconda\envs\test\lib\site-packages\_pytest\capture.py", line 900, in pytest_runtest_call
    return (yield)
  File "C:\Miniconda\envs\test\lib\site-packages\pluggy\_callers.py", line 139, in _multicall
    teardown.throw(exception)
  File "C:\Miniconda\envs\test\lib\site-packages\_pytest\skipping.py", line 268, in pytest_runtest_call
    return (yield)
  File "C:\Miniconda\envs\test\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Miniconda\envs\test\lib\site-packages\_pytest\runner.py", line 179, in pytest_runtest_call
    item.runtest()
  File "C:\Miniconda\envs\test\lib\site-packages\_pytest\python.py", line 1720, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "C:\Miniconda\envs\test\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Miniconda\envs\test\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Miniconda\envs\test\lib\site-packages\pluggy\_callers.py", line 167, in _multicall
    raise exception
  File "C:\Miniconda\envs\test\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Miniconda\envs\test\lib\site-packages\_pytest\python.py", line 166, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "D:\a\conda-build\conda-build\tests\test_api_build.py", line 523, in test_cmake_generator
    api.build(os.path.join(metadata_dir, "_cmake_generator"), config=testing_config)
  File "D:\a\conda-build\conda-build\conda_build\api.py", line 214, in build
    return build_tree(
  File "D:\a\conda-build\conda-build\conda_build\build.py", line 3687, in build_tree
    packages_from_this = build(
  File "D:\a\conda-build\conda-build\conda_build\build.py", line 2545, in build
    raise BuildScriptException(str(exc), caused_by=exc) from exc
conda_build.exceptions.BuildScriptException: Command '['cmd.exe', '/d', '/c', 'conda_build.bat']' returned non-zero exit status 1.
---------------------------- Captured stdout call -----------------------------
Attempting to finalize metadata for conda-build-test-cmake-generator
Reloading output folder: ...working... done
Getting pinned dependencies: ...working... done
BUILD START: ['conda-build-test-cmake-generator-1.0-h82bb817_0.conda']

## Package Plan ##

  environment location: D:\a\_temp\popen-gw1\test_cmake_generator_3_10_64_0\conda-build-test-cmake-generator_1781465578707\_h_env


The following NEW packages will be INSTALLED:

    ucrt:           10.0.22621.0-haa95532_0
    vc:             14.3-h2df5915_12       
    vc14_runtime:   14.44.35208-h4927774_12
    vs2015_runtime: 14.44.35208-ha6b5a95_12

Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
Reloading output folder: ...working... done
Solving environment (_build_env): ...working... done

## Package Plan ##

  environment location: D:\a\_temp\popen-gw1\test_cmake_generator_3_10_64_0\conda-build-test-cmake-generator_1781465578707\_build_env


The following NEW packages will be INSTALLED:

    bzip2:          1.0.8-h2bbff1b_6        
    cmake:          4.2.3-h93052ac_0        
    libuv:          1.52.0-hd0d7782_0       
    libzlib:        1.3.2-h1c6eee0_0        
    lz4-c:          1.9.4-h2bbff1b_1        
    ucrt:           10.0.22621.0-haa95532_0 
    vc:             14.3-h2df5915_12        
    vc14_runtime:   14.44.35208-h4927774_12 
    vs2015_runtime: 14.44.35208-ha6b5a95_12 
    vs2017_win-64:  19.16.27032.1-hb4161e2_3
    vswhere:        3.1.7-haa95532_0        
    xz:             5.8.2-h53af0af_0        
    zlib:           1.3.2-h1c6eee0_0        
    zstd:           1.5.7-h56299aa_0        

Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done

Rendered as:
```yaml
package:
  name: conda-build-test-cmake-generator
  version: '1.0'
build: {}
requirements:
  build:
    - ucrt 10.0.22621.0 haa95532_0
    - lz4-c 1.9.4 h2bbff1b_1
    - vs2017_win-64 19.16.27032.1 hb4161e2_3
    - vc14_runtime 14.44.35208 h4927774_12
    - zlib 1.3.2 h1c6eee0_0
    - libuv 1.52.0 hd0d7782_0
    - vs2015_runtime 14.44.35208 ha6b5a95_12
    - libzlib 1.3.2 h1c6eee0_0
    - cmake 4.2.3 h93052ac_0
    - vc 14.3 h2df5915_12
    - xz 5.8.2 h53af0af_0
    - zstd 1.5.7 h56299aa_0
    - bzip2 1.0.8 h2bbff1b_6
    - vswhere 3.1.7 haa95532_0
  host:
    - ucrt 10.0.22621.0 haa95532_0
    - vc14_runtime 14.44.35208 h4927774_12
    - vc 14.3 h2df5915_12
    - vs2015_runtime 14.44.35208 ha6b5a95_12
  run:
    - vc >=14.1,<15.0a0
    - vs2015_runtime >=14.16.27012,<15.0a0
about: {}
extra:
  final: true
  copy_test_source_files: true
```

source tree in: D:\a\_temp\popen-gw1\test_cmake_generator_3_10_64_0\conda-build-test-cmake-generator_1781465578707\work
Rewriting env in output: {'BUILD_PREFIX': 'D:\\a\\_temp\\popen-gw1\\test_cmake_generator_3_10_64_0\\conda-build-test-cmake-generator_1781465578707\\_build_env',
 'PREFIX': 'D:\\a\\_temp\\popen-gw1\\test_cmake_generator_3_10_64_0\\conda-build-test-cmake-generator_1781465578707\\_h_env',
 'SRC_DIR': 'D:\\a\\_temp\\popen-gw1\\test_cmake_generator_3_10_64_0\\conda-build-test-cmake-generator_1781465578707\\work'}

%SRC_DIR%>IF "" == "" (call %SRC_DIR%\build_env_setup.bat ) 

%SRC_DIR%>set "CONDA_BUILD=1" 

%SRC_DIR%>set "PYTHONNOUSERSITE=1" 

%SRC_DIR%>set "CONDA_DEFAULT_ENV=%PREFIX%" 

%SRC_DIR%>set "ARCH=64" 

%SRC_DIR%>set "PREFIX=%PREFIX%" 

%SRC_DIR%>set "BUILD_PREFIX=%BUILD_PREFIX%" 

%SRC_DIR%>set "SYS_PREFIX=C:\Miniconda\envs\test" 

%SRC_DIR%>set "SYS_PYTHON=C:\Miniconda\envs\test\python.exe" 

%SRC_DIR%>set "SUBDIR=win-64" 

%SRC_DIR%>set "build_platform=win-64" 

%SRC_DIR%>set "SRC_DIR=%SRC_DIR%" 

%SRC_DIR%>set "ROOT=C:\Miniconda\envs\test" 

%SRC_DIR%>set "CONDA_PY=310" 

%SRC_DIR%>set "PY3K=1" 

%SRC_DIR%>set "PY_VER=3.10" 

%SRC_DIR%>set "STDLIB_DIR=%PREFIX%\Lib" 

%SRC_DIR%>set "SP_DIR=%PREFIX%\Lib\site-packages" 

%SRC_DIR%>set "NPY_VER=1.22" 

%SRC_DIR%>set "CONDA_NPY=1.22" 

%SRC_DIR%>set "NPY_DISTUTILS_APPEND_FLAGS=1" 

%SRC_DIR%>set "PERL_VER=5.26" 

%SRC_DIR%>set "CONDA_PERL=5.26.2" 

%SRC_DIR%>set "LUA_VER=5" 

%SRC_DIR%>set "CONDA_LUA=5" 

%SRC_DIR%>set "R_VER=3.4" 

%SRC_DIR%>set "CONDA_R=3.4" 

%SRC_DIR%>set "PKG_NAME=conda-build-test-cmake-generator" 

%SRC_DIR%>set "PKG_VERSION=1.0" 

%SRC_DIR%>set "PKG_BUILDNUM=0" 

%SRC_DIR%>set "PKG_BUILD_STRING=placeholder" 

%SRC_DIR%>set "PKG_HASH=1234567" 

%SRC_DIR%>set "RECIPE_DIR=D:\a\conda-build\conda-build\tests\test-recipes\metadata\_cmake_generator" 

%SRC_DIR%>set "CPU_COUNT=4" 

%SRC_DIR%>set "SHLIB_EXT=.dll" 

%SRC_DIR%>set "PATH=%BUILD_PREFIX%;%BUILD_PREFIX%\Library\mingw-w64\bin;%BUILD_PREFIX%\Library\usr\bin;%BUILD_PREFIX%\Library\bin;%BUILD_PREFIX%\Scripts;%BUILD_PREFIX%\bin;%PREFIX%;%PREFIX%\Library\mingw-w64\bin;%PREFIX%\Library\usr\bin;%PREFIX%\Library\bin;%PREFIX%\Scripts;%PREFIX%\bin;C:\Miniconda\envs\test;C:\Miniconda\envs\test\Library\mingw64\bin;C:\Miniconda\envs\test\Library\mingw-w64\bin;C:\Miniconda\envs\test\Library\usr\bin;C:\Miniconda\envs\test\Library\bin;C:\Miniconda\envs\test\Scripts;C:\Miniconda\envs\test\bin;C:\Program Files\PowerShell\7;C:\Miniconda\condabin;C:\Program Files\MongoDB\Server\7.0\bin;C:\vcpkg;C:\tools\zstd;C:\hostedtoolcache\windows\stack\3.9.3\x64;C:\cabal\bin;C:\ghcup\bin;C:\mingw64\bin;C:\Program Files\dotnet;C:\Program Files\MySQL\MySQL Server 8.0\bin;C:\Program Files\R\R-4.6.0\bin\x64;C:\SeleniumWebDrivers\GeckoDriver;C:\SeleniumWebDrivers\EdgeDriver;C:\SeleniumWebDrivers\ChromeDriver;C:\Program Files (x86)\sbt\bin;C:\Program Files (x86)\GitHub CLI;C:\Program Files\Git\bin;C:\Program Files (x86)\pipx_bin;C:\npm\prefix;C:\hostedtoolcache\windows\go\1.24.13\x64\bin;C:\hostedtoolcache\windows\Python\3.12.10\x64\Scripts;C:\hostedtoolcache\windows\Python\3.12.10\x64;C:\hostedtoolcache\windows\Ruby\3.3.11\x64\bin;C:\Program Files\OpenSSL\bin;C:\tools\kotlinc\bin;C:\hostedtoolcache\windows\Java_Temurin-Hotspot_jdk\17.0.19-10\x64\bin;C:\Program Files\ImageMagick-7.1.2-Q16-HDRI;C:\Program Files\Microsoft SDKs\Azure\CLI2\wbin;C:\ProgramData\kind;C:\ProgramData\Chocolatey\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0;C:\Windows\System32\OpenSSH;C:\Program Files\PowerShell\7;C:\Program Files\Microsoft\Web Platform Installer;C:\Program Files\Microsoft SQL Server\170\Tools\Binn;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn;C:\Program Files\dotnet;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit;C:\Program Files (x86)\WiX Toolset v3.14\bin;C:\Program Files\Microsoft SQL Server\130\DTS\Binn;C:\Program Files\Microsoft SQL Server\140\DTS\Binn;C:\Program Files\Microsoft SQL Server\150\DTS\Binn;C:\Program Files\Microsoft SQL Server\160\DTS\Binn;C:\Program Files\Microsoft SQL Server\170\DTS\Binn;C:\ProgramData\chocolatey\lib\pulumi\tools\Pulumi\bin;C:\Program Files\CMake\bin;C:\Strawberry\c\bin;C:\Strawberry\perl\site\bin;C:\Strawberry\perl\bin;C:\ProgramData\chocolatey\lib\maven\apache-maven-3.9.16\bin;C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code;C:\Program Files\Microsoft SDKs\Service Fabric\Tools\ServiceFabricLocalClusterManager;C:\Program Files\nodejs;C:\Program Files\Git\cmd;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\Program Files\GitHub CLI;c:\tools\php;C:\Program Files (x86)\sbt\bin;C:\Program Files\Amazon\AWSCLIV2;C:\Program Files\Amazon\SessionManagerPlugin\bin;C:\Program Files\Amazon\AWSSAMCLI\bin;C:\Program Files\Microsoft SQL Server\130\Tools\Binn;C:\Program Files\mongosh;C:\Program Files\LLVM\bin;C:\Program Files (x86)\LLVM\bin;C:\Users\runneradmin\.dotnet\tools;C:\Users\runneradmin\.cargo\bin;C:\Users\runneradmin\AppData\Local\Microsoft\WindowsApps" 

%SRC_DIR%>set "SCRIPTS=%PREFIX%\Scripts" 

%SRC_DIR%>set "LIBRARY_PREFIX=%PREFIX%\Library" 

%SRC_DIR%>set "LIBRARY_BIN=%PREFIX%\Library\bin" 

%SRC_DIR%>set "LIBRARY_INC=%PREFIX%\Library\include" 

%SRC_DIR%>set "LIBRARY_LIB=%PREFIX%\Library\lib" 

%SRC_DIR%>set "CYGWIN_PREFIX=/cygdrive/d/a/_temp/popen-gw1/test_cmake_generator_3_10_64_0/conda-build-test-cmake-generator_1781465578707/_h_env" 

%SRC_DIR%>set "ALLUSERSPROFILE=C:\ProgramData" 

%SRC_DIR%>set "APPDATA=C:\Users\runneradmin\AppData\Roaming" 

%SRC_DIR%>set "CommonProgramFiles=C:\Program Files\Common Files" 

%SRC_DIR%>set "CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files" 

%SRC_DIR%>set "CommonProgramW6432=C:\Program Files\Common Files" 

%SRC_DIR%>set "COMPUTERNAME=runnervmhv6ak" 

%SRC_DIR%>set "ComSpec=C:\Windows\system32\cmd.exe" 

%SRC_DIR%>set "HOMEDRIVE=C:" 

%SRC_DIR%>set "HOMEPATH=\Users\runneradmin" 

%SRC_DIR%>set "LOCALAPPDATA=C:\Users\runneradmin\AppData\Local" 

%SRC_DIR%>set "LOGONSERVER=\\runnervmhv6ak" 

%SRC_DIR%>set "NUMBER_OF_PROCESSORS=4" 

%SRC_DIR%>set "PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.CPL" 

%SRC_DIR%>set "ProgramData=C:\ProgramData" 

%SRC_DIR%>set "ProgramFiles=C:\Program Files" 

%SRC_DIR%>set "ProgramFiles(x86)=C:\Program Files (x86)" 

%SRC_DIR%>set "ProgramW6432=C:\Program Files" 

%SRC_DIR%>set "PSModulePath=C:\Users\runneradmin\Documents\PowerShell\Modules;C:\Program Files\PowerShell\Modules;c:\program files\powershell\7\Modules;C:\\Modules\az_15.6.1;C:\Users\packer\Documents\WindowsPowerShell\Modules;C:\Program Files\WindowsPowerShell\Modules;C:\Windows\system32\WindowsPowerShell\v1.0\Modules;C:\Program Files\Microsoft SQL Server\130\Tools\PowerShell\Modules\" 

%SRC_DIR%>set "PUBLIC=C:\Users\Public" 

%SRC_DIR%>set "SystemDrive=C:" 

%SRC_DIR%>set "SystemRoot=C:\Windows" 

%SRC_DIR%>set "TEMP=C:\Users\RUNNER~1\AppData\Local\Temp" 

%SRC_DIR%>set "TMP=C:\Users\RUNNER~1\AppData\Local\Temp" 

%SRC_DIR%>set "USERDOMAIN=runnervmhv6ak" 

%SRC_DIR%>set "USERNAME=runneradmin" 

%SRC_DIR%>set "USERPROFILE=C:\Users\runneradmin" 

%SRC_DIR%>set "windir=C:\Windows" 

%SRC_DIR%>set "PROCESSOR_ARCHITECTURE=AMD64" 

%SRC_DIR%>set "PROCESSOR_IDENTIFIER=AMD64 Family 25 Model 17 Stepping 1, AuthenticAMD" 

%SRC_DIR%>set "BUILD=amd64-pc-windows-19.0.0" 

%SRC_DIR%>set "extend_keys={'extend_keys', 'ignore_build_only_deps', 'pin_run_as_build', 'ignore_version'}" 

%SRC_DIR%>set "ignore_build_only_deps={'numpy', 'python'}" 

%SRC_DIR%>set "pin_run_as_build=OrderedDict([('python', {'min_pin': 'x.x', 'max_pin': 'x.x'}), ('r-base', {'min_pin': 'x.x', 'max_pin': 'x.x'})])" 

%SRC_DIR%>set "cpu_optimization_target=nocona" 

%SRC_DIR%>set "cran_mirror=https://cran.r-project.org" 

%SRC_DIR%>set "fortran_compiler=gfortran" 

%SRC_DIR%>set "vc=14" 

%SRC_DIR%>set "target_platform=win-64" 

%SRC_DIR%>set "cxx_compiler=vs2017" 

%SRC_DIR%>set "r_base=3.4" 

%SRC_DIR%>set "c_compiler=vs2017" 

%SRC_DIR%>set "CONDA_BUILD_STATE=BUILD" 

%SRC_DIR%>set "PIP_NO_BUILD_ISOLATION=False" 

%SRC_DIR%>set "PIP_NO_DEPENDENCIES=True" 

%SRC_DIR%>set "PIP_IGNORE_INSTALLED=True" 

%SRC_DIR%>set "PIP_CACHE_DIR=D:\a\_temp\popen-gw1\test_cmake_generator_3_10_64_0\conda-build-test-cmake-generator_1781465578707\pip_cache" 

%SRC_DIR%>set "PIP_NO_INDEX=True" 

%SRC_DIR%>set "CMAKE_GENERATOR=Visual Studio 17 2022" 

%SRC_DIR%>set "INCLUDE=%PREFIX%\Library\include;" 

%SRC_DIR%>set "LIB=%PREFIX%\Library\lib;" 

%SRC_DIR%>call "C:\Miniconda\envs\test\condabin\conda_hook.bat" 

%SRC_DIR%>set "CONDA_EXE=C:\Miniconda\envs\test\Scripts\conda.exe" 

%SRC_DIR%>set "_CONDA_EXE=C:\Miniconda\envs\test\Scripts\conda.exe" 

%SRC_DIR%>set "_CE_M=" 

%SRC_DIR%>set "_CE_CONDA=" 

%SRC_DIR%>set "CONDA_PYTHON_EXE=C:\Miniconda\envs\test\python.exe" 

%SRC_DIR%>set "_CONDA_ROOT=C:\Miniconda\envs\test" 

%SRC_DIR%>call "C:\Miniconda\envs\test\condabin\conda.bat" activate --stack "%BUILD_PREFIX%" 

%SRC_DIR%>SET DISTUTILS_USE_SDK=1 

%SRC_DIR%>SET MSSdk=1 

%SRC_DIR%>SET "VS_VERSION=15.0" 

%SRC_DIR%>SET "VS_MAJOR=15" 

%SRC_DIR%>SET "VS_YEAR=2017" 

%SRC_DIR%>set "MSYS2_ARG_CONV_EXCL=/AI;/AL;/OUT;/out" 

%SRC_DIR%>set "MSYS2_ENV_CONV_EXCL=CL" 

%SRC_DIR%>set "PY_VCRUNTIME_REDIST=%PREFIX%\bin\vcruntime140.dll" 

%SRC_DIR%>set "CXX=cl.exe" 

%SRC_DIR%>set "CC=cl.exe" 

%SRC_DIR%>CALL %BUILD_PREFIX%\etc\conda\activate.d\vs2017_get_vsinstall_dir.bat 
Windows SDK version found as: "10.0.26100.0"
**********************************************************************
** Visual Studio 2017 Developer Command Prompt v15.0
** Copyright (c) 2017 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'
C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\bin\Hostx64\x64\cl.exe
C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\bin\Hostx64\x64\link.exe
C:\Miniconda\envs\test\Library\usr\bin\link.exe
C:\Program Files\Git\usr\bin\link.exe
C:\mingw64\bin\gcc.exe
C:\Strawberry\c\bin\gcc.exe
-- Configuring incomplete, errors occurred!
---------------------------- Captured stderr call -----------------------------
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


CMake Error at CMakeLists.txt:3 (PROJECT):
  Generator

    Visual Studio 17 2022

  could not find any instance of Visual Studio.



CMake Error:
  Generator

    Visual Studio 17 2022

  could not find any instance of Visual Studio.



------------------------------ Captured log call ------------------------------
INFO     conda_build.metadata:metadata.py:1037 Attempting to finalize metadata for conda-build-test-cmake-generator
``````

--- a/conda_build/windows.py
+++ b/conda_build/windows.py
@@ -41,6 +41,18 @@ VS_VERSION_STRING = {
     "17.0": "Visual Studio 17 2022",
 }
 
+# Map from conda compiler package name to VS version string key
+VS_COMPILER_VERSION = {
+    "vs2008": "9.0",
+    "vs2010": "10.0",
+    "vs2012": "11.0",
+    "vs2013": "12.0",
+    "vs2015": "14.0",
+    "vs2017": "15.0",
+    "vs2019": "16.0",
+    "vs2022": "17.0",
+}
+
 
 def fix_staged_scripts(scripts_dir, config):
     """
@@ -307,32 +319,32 @@ def write_build_scripts(m, env, bld_bat):
                     override=m.get_value("build/msvc_compiler", None),
                 )
             )
-        else:
-            # Set CMAKE_GENERATOR for new-style compiler activation
-            version = m.get_value("build/msvc_compiler", None)
-            if not version:
-                from .variants import get_default_variant
-
-                py_ver = m.config.variant.get(
-                    "python", get_default_variant(m.config)["python"]
-                )
-                py_ver = tuple([int(x) for x in py_ver.split(".")[:2]])
-                if py_ver >= (3, 6):
-                    version = "17.0"
-                elif py_ver == (3, 5):
-                    version = "15.0"
-                elif py_ver >= (3, 0):
-                    version = "10.0"
-                else:  # 2.x
-                    version = "9.0"
-            if version:
-                fo.write(f'set "CMAKE_GENERATOR={VS_VERSION_STRING[version]}"\n')
         # Reset echo on, because MSVC scripts might have turned it off
         fo.write("@echo on\n")
         fo.write('set "INCLUDE={};%INCLUDE%"\n'.format(env["LIBRARY_INC"]))
         fo.write('set "LIB={};%LIB%"\n'.format(env["LIBRARY_LIB"]))
         if m.config.activate and m.name() != "conda":
             write_bat_activation_text(fo, m)
+        if m.uses_new_style_compiler_activation:
+            # Set CMAKE_GENERATOR after conda activation so it takes precedence over
+            # any CMAKE_GENERATOR set by the activated compiler package's activation
+            # scripts (e.g., vs2017 may set "Visual Studio 15 2017 Win64" which is no
+            # longer valid in CMake 4.0+, which removed the platform suffix).
+            # Use the same compiler that the recipe will use (from variant or default).
+            msvc_override = m.get_value("build/msvc_compiler", None)
+            if msvc_override and msvc_override in VS_VERSION_STRING:
+                # Legacy msvc_compiler key: already a version string like "15.0"
+                fo.write(f'set "CMAKE_GENERATOR={VS_VERSION_STRING[msvc_override]}"\n')
+            else:
+                from .jinja_context import native_compiler
+
+                compiler_name = m.config.variant.get(
+                    "c_compiler",
+                    native_compiler("c", m.config),
+                )
+                version = VS_COMPILER_VERSION.get(compiler_name)
+                if version:
+                    fo.write(f'set "CMAKE_GENERATOR={VS_VERSION_STRING[version]}"\n')
     # bld_bat may have been generated elsewhere with contents of build/script
     work_script = join(m.config.work_dir, "conda_build.bat")
     if os.path.isfile(bld_bat):

--- a/news/5829-fix-cmake-generator-windows.md
+++ b/news/5829-fix-cmake-generator-windows.md
@@ -1,0 +1,19 @@
+### Enhancements
+
+* <news item>
+
+### Bug fixes
+
+* Fix `CMAKE_GENERATOR` being overwritten by conda compiler package activation scripts on Windows, which caused builds to fail with CMake 4.0+ due to the removal of the `Win64` platform suffix from Visual Studio generator names. The generator is now set after conda activation and is derived from the actual compiler package being used (e.g., `vs2017` → `Visual Studio 15 2017`). (#5829)
+
+### Deprecations
+
+* <news item>
+
+### Docs
+
+* <news item>
+
+### Other
+
+* <news item>

--- a/news/5829-fix-cmake-generator-windows.md
+++ b/news/5829-fix-cmake-generator-windows.md
@@ -4,7 +4,7 @@
 
 ### Bug fixes
 
-* Fix `CMAKE_GENERATOR` being overwritten by conda compiler package activation scripts on Windows, which caused builds to fail with CMake 4.0+ due to the removal of the `Win64` platform suffix from Visual Studio generator names. The generator is now set after conda activation and is derived from the actual compiler package being used (e.g., `vs2017` → `Visual Studio 15 2017`). (#5829)
+* Fix `CMAKE_GENERATOR` being overwritten by conda compiler package activation scripts on Windows, which caused builds to fail with CMake 4.0+ due to the removal of the `Win64` platform suffix from Visual Studio generator names. The generator is now set after conda activation and is derived from the actual compiler package being used (e.g., `vs2017` → `Visual Studio 15 2017`). (#6013 via #6012)
 
 ### Deprecations
 

